src/odemis/driver/spectrometer.py

```diff
@@ -295,16 +295,20 @@ def _newFrame(self, df, data):
             # Subtract baseline (aka black level) to avoid it from being multiplied
             try:
                 baseline = data.metadata[model.MD_BASELINE]
-                baseline_sum = orig_shape[1] * baseline
+                baseline_sum = orig_shape[0] * baseline
                 # If the baseline is too high compared to the actual black, we
                 # could end up subtracting too much, and values would underflow
                 # => be extra careful and never subtract more than min value.
-                extra_bl = min(int(data.min()), baseline_sum) - baseline
-                if extra_bl > 0:
-                    data -= extra_bl
-                else:
-                    logging.info("Baseline reported at %d, but lower values found, so not compensating",
-                                 baseline)
+                minv = float(data.min())
+                extra_bl = baseline_sum - baseline
```

Comment:
I don't get this part. So `baseline_sum` is just the baseline value times the shape of the data, so you get an 1dim array with values of baseline. Then you subtract from this array values again exactly this value, which would be 0 everywhere. What do I miss?

Reply:
No `baseline_sum` is just one value: the baseline value after binning (summing) the vertical axis. We want to reduce this "black level" back to the standard baseline => so extra_bl contains a single value, which is how much we should subtract to get back to the standard baseline.

Reply:
Ah ok: So `extra_bl = (orig_shape[0] - 1) * baseline`.
And data is already binned?
I still don't get why you subtract  `(orig_shape[0] - 1) * baseline` instead of  `orig_shape[0] * baseline`.